src/main/java/com/capstone/doctorsOffice/services/AppointmentServiceImpl.java

```diff
@@ -0,0 +1,66 @@
+package com.capstone.doctorsOffice.services;
+
+import com.capstone.doctorsOffice.dtos.*;
+import com.capstone.doctorsOffice.entities.*;
+import com.capstone.doctorsOffice.repositories.*;
+import jakarta.transaction.Transactional;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+import java.util.stream.Collectors;
+
+@Service
+public class AppointmentServiceImpl implements AppointmentService {
+    @Autowired
+    private DoctorRepository doctorRepository;
+    @Autowired
+    private AppointmentRepository appointmentRepository;
+    @Autowired
+    private PatientRepository patientRepository;
+
+    @Override
+    @Transactional
+    public List<String> addNewAppointment(AppointmentDto appointmentDto){
+        List<String> response = new ArrayList<>();
+        Appointment appointment = new Appointment(appointmentDto);
+        appointmentRepository.saveAndFlush(appointment);
+        response.add("New Appointment Scheduled");
+        return response;
+    }
+
+    @Override
+    @Transactional
+    public void deleteAppointment(Long appointmentId){
+        Optional<Appointment> appointmentOptional = appointmentRepository.findById(appointmentId);
+        appointmentOptional.ifPresent(appointment -> appointmentRepository.delete(appointment));
+    }
+
+    @Override
+    @Transactional
+    public void updateAppointment(Long appointmentId, AppointmentDto appointmentDto){
+        Optional<Appointment> appointmentOptional = appointmentRepository.findById(appointmentDto.getAppointmentId());
+        appointmentOptional.ifPresent(appointment -> {
```

Comment:
We need to use the `appointmentId` to look up the entity. The Dot does not have the ID.
should be:
`Optional<Appointment> appointmentOptional = appointmentRepository.findById(appointmentId);`

Reply:
done